src/main/scala/strawman/collection/mutable/ArrayDeque.scala

```diff
@@ -0,0 +1,292 @@
+package scala
+package collection.mutable
```

Comment:
But if you put a class in package `scala.collection.mutable`, it belongs in directory `scala/collection/mutable`.

But more fundamentally, IIUC collections in this repo don't use `scala.collection` at all, since the point of this repo is to completely redesign that package for 2.13. @julienrf am I missing something?
I don't know if this should be ported to the 2.13 API, but it seems that should be clarified *before* focusing on low-level details.

I don't know how much effort it'd be to port this to `collection.strawman`, but it's probably non-trivial.

Because of the binary compatibility constraints, this code cannot be added to 2.12.x (and this would have been the wrong repo anyway) as long as it changes the API (even adding methods/classes is forbidden, because of forward compatibility—clients built with 2.12.2 must run against 2.12.0). It can be shipped as an external library for 2.12.x users, though it's maybe better then to put it in another package and give up on the optimization (adding to an existing package from a new JAR is forbidden by OSGi).

Because the APIs you're using will change a lot for 2.13, this code can't be merged as-is in 2.13 either.

Not my decision of course—it's just my understanding of the existing architectural rules.